radlab-ui/webapp/src/pages/api/secret.ts

```diff
@@ -0,0 +1,86 @@
+import { NextApiRequest, NextApiResponse } from "next"
+import { envOrFail } from "@/utils/env"
+import { SecretManagerServiceClient } from "@google-cloud/secret-manager"
+import { ISecretManagerReq } from "@/utils/types"
+
+const secretmanagerClient = new SecretManagerServiceClient()
+
+const gcpProjectId = envOrFail(
+  "NEXT_PUBLIC_GCP_PROJECT_ID",
+  process.env.NEXT_PUBLIC_GCP_PROJECT_ID,
+)
+
+export const getSecretKeyValue = async (secretId: string) => {
+  try {
+    const parent = `projects/${gcpProjectId}`
+    const request = {
+      name: `${parent}/secrets/${secretId}/versions/latest`,
+    }
+    const [response] = await secretmanagerClient.accessSecretVersion(request)
+    const value = response.payload?.data?.toString()
+    return value
+  } catch (error) {
+    console.error(error)
+    return null
+  }
+}
+
+const createSecret = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { key: secretId, value }: ISecretManagerReq = req.body
+
+  const parent = `projects/${gcpProjectId}`
+  const secretKeyName = `${parent}/secrets/${secretId}`
+
+  const request = {
+    parent,
+    secretId,
+    secret: {
+      replication: {
+        userManaged: {
+          replicas: [
+            {
+              location: "us-central1",
```

Comment:
Do we need this region? If so, it can't be hard coded. It needs to be set as an environment variable and documented here: https://googlecloudplatform.github.io/rad-lab/docs/rad-lab-ui/ui_installation/env-variables/

Reply:
Yes its required hence storing it as env and updated the document.